boards/adafruit-itsybitsy-m4/board.c

```diff
@@ -54,7 +54,7 @@ static mtd_spi_nor_t samd51_nor_dev = {
     .params = &_samd51_nor_params,
 };
 
-mtd_dev_t *mtd0 = (mtd_dev_t *)&samd51_nor_dev;
+XFA(mtd_dev_xfa, 0) mtd_dev_t *mtd0 = (mtd_dev_t *)&samd51_nor_dev;
```

Comment:
This is kind of ugly to have in each board - how about a macro

```C
#define MTD_XFA_ADD(dev, idx) XFA(mtd_dev_xfa, (idx)) mtd_dev_t *mtd # (idx) = (mtd_dev_t *)&(dev);
```

Then this would be

```suggestion
MTD_XFA_ADD(samd51_nor_dev, 0);
```

Reply:
The `0` in `XFA(mtd_dev_xfa, 0)` is not the index but the priority. But defining the macro as
```
define MTD_XFA_ADD(dev, idx) XFA(mtd_dev_xfa, 0) mtd_dev_t *mtd # (idx) = (mtd_dev_t *)&(dev);
```
seems to be reasonable.

Reply:
@benpicco I have added the macro according to your suggestion. Parameter `idx` is used now as priority so that the index corresponds to the position in the XFA provided that the indices are unique, start from 0 and are consecutive.